src/GraphQL/Types/Composite/AutoRegisteringObjectGraphType.cs

```diff
@@ -77,7 +78,89 @@ protected virtual IEnumerable<FieldType> ProvideFields()
         {
             var typeInformation = GetTypeInformation(memberInfo);
             var graphType = typeInformation.ConstructGraphType();
-            return AutoRegisteringHelper.CreateField(memberInfo, graphType, false);
+            var fieldType = AutoRegisteringHelper.CreateField(memberInfo, graphType, false);
+            if (memberInfo is MethodInfo methodInfo)
+            {
+                if (fieldType.Name.EndsWith("Async") && methodInfo.ReturnType.IsGenericType && methodInfo.ReturnType.GetGenericTypeDefinition() == typeof(Task<>))
+                {
+                    fieldType.Name = fieldType.Name.Substring(0, fieldType.Name.Length - 5);
+                }
+                var (queryArguments, resolver) = BuildMethodResolver(methodInfo, fieldType);
+                fieldType.Arguments = queryArguments;
+                fieldType.Resolver = resolver;
+            }
+            else
+            {
+                fieldType.Resolver = MemberResolver.Create(memberInfo);
+            }
+            // apply field attributes after resolver has been set
+            AutoRegisteringHelper.ApplyFieldAttributes(memberInfo, fieldType, false);
+            return fieldType;
+        }
+
+        /// <summary>
+        /// Retuns a list of query arguments within a <see cref="QueryArguments"/> instance, and a
+        /// <see cref="IFieldResolver"/> instance, for the specified field.
+        /// </summary>
+        protected (QueryArguments? QueryArguments, IFieldResolver Resolver) BuildMethodResolver(MethodInfo methodInfo, FieldType fieldType)
```

Comment:
This is not `virtual`.  I cannot think of any reason it should be virtual, and it is quite complex code that isn't easily replicated.  Any behavior that anyone would want to change has been moved to the `GetArgumentInformation` and `ApplyArgumentAttributes` virtual methods.  By overriding `CreateField`, which contains the only call to `BuildMethodResolver`, it is possible to "override" this method also.

Reply:
ok